hoomd/python_action.py

```diff
@@ -0,0 +1,79 @@
+from hoomd.operation import _TriggeredOperation
+from hoomd.parameterdicts import ParameterDict
+from hoomd.custom_action import _CustomAction
+from hoomd.typeconverter import OnlyType
+from hoomd.trigger import Trigger
+from hoomd.util import trigger_preprocessing
+from hoomd.logger import LoggerQuantity
+from hoomd import _hoomd
+
+
+class _PythonAction(_TriggeredOperation):
```

Comment:
We need clearer names for these, it is hard to keep track of what goes where. Based on the current words we are using, we have Operations, Actions, Analyzers, Updaters, Integrators, Tuners, and Computes.

Operations are Python side objects which are responsible for providing UI and documentation. Each operation is composed of an Action which is responsible for actually performing the "action" of the "operation".

Both operations and actions are further grouped into Analyzers, Updaters, Integrators, Tuners, and Computes. Renaming C++ classes is outside the scope of this PR, but maybe we can decide now on what those names will be and start using them for the custom objects?

The double use of Analyzers, Updaters, Integrators, Tuners, and Computes creates confusion. One solution would be to have AnalyzerAction, UpdaterAction, IntegratorAction, TunerAction, and ComputeAction to distinguish them from the Analyzer that is an operation. An alternate thought is to have Action and TriggeredAction only and only distinguish between types of operations at the Python level.

Any thoughts?

Reply:
With regards to what we should call the objects the user creates, I think we could get away with just the `CustomAction` class for analyzers, updaters, and tuners. They could technically wrap it with any of the wrappers, but that is ultimately up to the user anyways.

The wrappers, I think make sense to call them `CustomUpdater`, `CustomAnalyzer`, etc. We need a different wrapper for each C++ class we allow.

I would also be fine with moving the Python class from `CustomAction` to just `Action` this would be shorter, and I don't think it would cause problems.

Assuming we extend this later we may need different names. For instance, a custom pair potential may need more thought. Possibilities are `PairPotentialWrapper` and `CustomPairPotential`, `CustomPairPotential` and `CustomPairPotentialAction`.